posthog/api/hog_function.py

```diff
@@ -154,6 +154,16 @@ def validate(self, attrs):
         hog_type = attrs.get("type", instance.type if instance else "destination")
         is_create = self.context.get("view") and self.context["view"].action == "create"
 
+        # If transformations are disabled and this is an update to a transformation, preserve original hog code
+        if (
+            not is_create
+            and not settings.HOG_TRANSFORMATIONS_ENABLED
+            and "hog" in attrs
+            and hog_type == "transformation"
+            and instance is not None
+        ):
+            attrs["hog"] = instance.hog  # Keep the original code
```

Comment:
I think I didn't make it clear what this needs. It should basically be:
```suggestion
        if hog_type == "transformation":
            if not template:
                raise serializers.ValidationError(
                    {"template_id": "Transformation functions must be created from a template."}
                )
            # Currently we do not allow modifying the core transformation templates
            attrs["hog"] = template.hog
            attrs["inputs_schema"] = template.inputs_schema
```

Reply:
You can then add the check for whether `HOG_TRANSFORMATIONS_CUSTOMIZATION_ENABLED` or something

Reply:
ah, so we also only allow transformations being created from templates for now 👍